components/ai_chat/core/browser/model_service.cc

```diff
@@ -113,6 +113,7 @@ const std::vector<mojom::ModelPtr>& GetLeoModels() {
       model->key = "chat-automatic";
       model->display_name = "Automatic";
       model->vision_support = true;
+      model->supports_tools = false;
```

Comment:
I think it's better to set this to true and have a small server side change to distribute requests to supported models if tools event present like we did for vision.

Reply:
Ok, it probably needs more thoughts on server side when triaging requests @Steeeephen to decide to use tools or not then. Not in the scope of this PR, just that limiting tools to a explicitly model choice would somehow defeat the purpose of having an automatic model.

Reply:
Hopefully there'll be a time very soon when all models will support tools

Reply:
Not 100% sure how it works on the browser/agentic browser side but in general I would think the less hardcoding we can do in the browser the better, dealing with things like this in the Leo backend lets us move quicker

Reply:
@Steeeephen is it possible for server (the triaging model) to decide whether to use tools or not based on the request when client always pass tools?